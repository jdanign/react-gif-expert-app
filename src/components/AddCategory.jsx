import {useState} from 'react';


export const AddCategory = ({onNewCategory}) => {
	const inputValueDefault = '';
	const [inputValue, setInputValue] = useState(inputValueDefault);


	// Establece el valor del estado al cambiar el input, pulsando teclas dentro por ejemplo
	const onInputChange = ({target})=>{
		setInputValue(target.value);
	}


	const onSubmit = (event)=>{
		const cleanValue = inputValue.trim();
		// Evita que el formulario se envíe y recargue la página
		event.preventDefault();

		// Si tiene más de un caracter y no está vacío
		if (cleanValue.length > 1)
			onNewCategory(cleanValue);
		
		// Limpia los caracteres del estado del input después de enviar el formulario
		setInputValue(inputValueDefault);
	}


	return (
		<form onSubmit={onSubmit}>
			<input
				type="text"
				placeholder="Buscar gifs"
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	)
}