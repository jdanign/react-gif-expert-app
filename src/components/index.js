// Si en el import solo se hace referencia a la carpeta se importar√° el index.js

export * from './AddCategory';
export * from './GifGrid';
export * from './GiftItem'