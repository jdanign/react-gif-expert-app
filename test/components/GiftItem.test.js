import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";

describe('Pruebas en <GiftItem/>', () => {
    const title = 'Hola';
    const url = 'https://www.one-punch.com/saitama.jpg';

    test('Debe hacer match con el snapshot', () => {
        const {container} = render(
            <GiftItem title={title} url={url}/>
        );

        expect(container).toMatchSnapshot();
    });


    test('Debe mostrar la imagen con la URL y el ALT indicados', () => {
        render(<GiftItem title={title} url={url}/>);
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url)
        expect(alt).toBe(title)
    });


    test('Debe mostrar el tÃ­tulo en el componente', () => {
        render(<GiftItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})