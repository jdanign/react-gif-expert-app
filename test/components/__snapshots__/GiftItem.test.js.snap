// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
      src="https://www.one-punch.com/saitama.jpg"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
